src/preact/component/intersection-observer.js

```diff
@@ -13,17 +10,18 @@ import {useCallback, useRef} from '#preact';
  */
 export function useIntersectionObserver(callback) {
   const nodeRef = useRef(null);
+  const unobserveRef = useRef(null);
   const refCb = useCallback(
     (node) => {
       const prevNode = nodeRef.current;
       nodeRef.current = node;
       if (prevNode) {
-        unobserveIntersections(prevNode, callback);
+        unobserveRef.current();
```

Comment:
Nit: split this to make it clear what `current` is expected to be:

```suggestion
        const cleanup = unobserveRef.current;
        cleanup();
```

Reply:
Done.